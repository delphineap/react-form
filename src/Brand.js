import React from 'react';

const Brand = (props) => (
    <div className="brand" onClick={props.click}>
        <h3><img src={props.data.src} alt="" height="50" /> {props.data.name}</h3>
    </div>
)

export default Brand;