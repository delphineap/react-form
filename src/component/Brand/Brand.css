.Brand {
    border: 1px solid #003300;
}