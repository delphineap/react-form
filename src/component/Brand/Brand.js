import React from 'react';
import PropTypes from 'prop-types';

import {
    withRouter
  } from 'react-router-dom'

class Brand extends React.Component {

    render () {
        return (
            <>
                <h3><img src={this.props.data.src} alt="" height="50" /> {this.props.data.name}</h3>
                <button onClick={() => { this.props.click() }}>Supprimer</button>
                <button onClick={() => { this.props.history.push(`/brands/${this.props.data.id}/edit`)}}>Modifier</button>
            </>
        )
    }
}

Brand.propTypes = {
    data: PropTypes.shape({
        src: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
    })
}

export default withRouter(Brand);