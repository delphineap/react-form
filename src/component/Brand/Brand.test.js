import React from 'react';

import Brand from './Brand';

import {render, fireEvent, cleanup} from 'react-testing-library'

afterEach(cleanup);

test('calls fonction on click', () => {
    const handleClick = jest.fn();

    const brand =
    {
        'id': 1,
        'name': 'test1',
        'src': 'test'
    }

    const {container} = render(<Brand data={brand} click={handleClick} />);

    const title = container.querySelector('h3');

    fireEvent.click(title);
    expect(handleClick).toHaveBeenCalled();
})