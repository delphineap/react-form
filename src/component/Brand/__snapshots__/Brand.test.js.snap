// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<h3
  onClick={[Function]}
>
  <img
    alt=""
    height="50"
  />
   
</h3>
`;
